kedro-datasets/kedro_datasets/snowflake/snowpark_dataset.py

```diff
@@ -2,16 +2,15 @@
 """
 import logging
 from copy import deepcopy
-from typing import Any, Dict, Union
+from typing import Any, Dict
 
-import pandas as pd
 import snowflake.snowpark as sp
 from kedro.io.core import AbstractDataSet, DataSetError
 
 logger = logging.getLogger(__name__)
 
 
-class SnowparkTableDataSet(AbstractDataSet[pd.DataFrame, pd.DataFrame]):
+class SnowparkTableDataSet(AbstractDataSet[sp.DataFrame, sp.DataFrame]):
```

Comment:
```suggestion
class SnowparkTableDataSet(AbstractDataSet):
```
I don't think we need this hint